pkg/providers/anthropic/chat_stream.go

```diff
@@ -2,15 +2,102 @@ package anthropic
 
 import (
 	"context"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
 
 	"glide/pkg/api/schemas"
 	"glide/pkg/providers/clients"
 )
 
+type Client struct {
+	APIKey string
+}
+
 func (c *Client) SupportChatStream() bool {
-	return false
+	return true
+}
+
+func (c *Client) ChatStream(ctx context.Context, chatReq *schemas.ChatRequest) (clients.ChatStream, error) {
+	apiURL := "https://api.anthropic.com/v1/complete"
+	requestBody := map[string]interface{}{
+		"model":                "claude-2",
+		"prompt":               chatReq.Message, // Assuming chatReq.Message contains the user's message
+		"max_tokens_to_sample": 256,
+		"stream":               true,
+	}
+	requestJSON, err := json.Marshal(requestBody)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", apiURL, bytes.NewBuffer(requestJSON))
+	if err != nil {
+		return nil, err
+	}
+	httpReq.Header.Set("Content-Type", "application/json")
+	httpReq.Header.Set("anthropic-version", "2023-06-01")
+	httpReq.Header.Set("x-api-key", c.APIKey)
+
+	resp, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	chatStream := &AnthropicChatStream{
+		responseBody: resp.Body,
+	}
+
+	return chatStream, nil
+}
+
+type AnthropicChatStream struct {
+	responseBody io.ReadCloser
+}
+
+func (s *AnthropicChatStream) Receive() (string, error) {
```

Comment:
Should be called `Recv()` to fulfil  [the ChatStream interface](https://github.com/EinStack/glide/blob/develop/pkg/providers/clients/stream.go#L7-L11):

```suggestion
func (s *AnthropicChatStream) Recv() (string, error) {
```

It also should return `(*schemas.ChatStreamChunk, error)` but I think you will get to that:

https://github.com/EinStack/glide/blob/develop/pkg/providers/clients/stream.go#L9